packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -131,11 +93,106 @@ function mapStandbyTransactionToFeedTokenTransaction(tx: StandbyTransaction): Fe
   }
 }
 
+function addFetchedTransactionFunction(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: any
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
 function TransactionFeed() {
+  const address = useSelector(walletAddressSelector)
```

Comment:
is it possible to abstract some logic away to a hook? in the current implementation we have `const { loading, error, transactions } = useQueryTransactionFeed()` - it'd be nice to keep something like this, so that this component can focus on the display part and not worry about coordinating the fetching of data/which page number is next. i'm not sure if you have already tried this and decided against it, i can see why the `pageInfo` part complicates things. thought i'd check anyway :)